Full paths to modules are being left in production output bundles

I'm working with the arcgis js api and webpack plugin which uses this plugin under the hood. 

I'm running into constant issues where sometimes after running a build, I'll run into errors where the workers fail to load. It doesn't happen all of the time, and it sometimes resolves itself for no apparent reason. I realize this might not even be related to your plugin, but if you have any ideas to offer, I would love to hear them. This is sort of a last ditch effort in fixing this reoccurring issue. 

Looks like this is the dojo config the esri webpack plugin uses:

These are the errors I'm getting:

## 1

```
VM53057:1 Access to XMLHttpRequest at 'file:///C:/Users/<username>/web/dl2-datafi/node_modules/arcgis-js-api/core/workers/worker.js' from origin 'http://lvh.me:9000' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, https.
```

## 2

```
[esri.core.workers] Failed to create Worker. Fallback to execute module in main thread 
t {name: "request:server", message: "Unable to load C:/Users/<username>/web/dl2-datafi-…es/arcgis-js-api/core/workers/worker.js status: 0", details: {…}}
details: {url: "C:/Users/<username>/web/dl2-datafi/node_modules/arcgis-js-api/core/workers/worker.js", requestOptions: {…}, getHeader: ƒ, httpStatus: 0, subCode: undefined, …}
message: "Unable to load C:/Users/<username>/web/dl2-datafi/node_modules/arcgis-js-api/core/workers/worker.js status: 0"
name: "request:server"
```

![image](https://user-images.githubusercontent.com/5471079/47927342-0462ad00-de91-11e8-9f7a-827d774cce6e.png)

## 3

```
Error: Module not found: C:/Users/<username>/web/dl2-datafi/node_modules/arcgis-js-api/views/2d/layers/features/Pipeline
```

![image](https://user-images.githubusercontent.com/5471079/47927305-e6954800-de90-11e8-9175-0da13c9f1c4f.png)

Theres a few concerns here: 
 - First the obvious one is my app won't load. That makes me sad.
 - Second the username of my account is actually getting put into the ouput build, which I'm not a big fan of either. 

If you have any thoughts, I'd love to hear them.

Packages | Version
-----------|---------
Webpack: | 4.24.0
Dojo-webpack-plugin: | 2.7.6
ArcGIS webpack plugin: |  4.9.0
Dojo: |  1.14.0
arcgis-js-api: | 4.9.3

Dojo.config: 
<details>

```json
{
    "hasCache": {
        "webpack": 1,
        "host-browser": 1,
        "dom": 1,
        "dojo-loader": 1,
        "dojo-has-api": 1,
        "dojo-dom-ready-api": 1,
        "dojo-sniff": 1,
        "dojo-test-sniff": 1,
        "config-deferredInstrumentation": false,
        "config-tlmSiblingOfDojo": 1,
        "dojo-config-api": 0,
        "esri-promise-compatibility": 1,
        "esri-webpack": 1,
        "dom-addeventlistener": true,
        "touch-events": false,
        "pointer-events": true,
        "device-width": 1536,
        "dom-attributes-explicit": true,
        "dom-attributes-specified-flag": false,
        "extend-dojo": 1,
        "dojo-guarantee-console": 1,
        "dojo-debug-messages": false,
        "dojo-modulePaths": 1,
        "dojo-moduleUrl": 1,
        "air": false,
        "webkit": 537.36,
        "chrome": 70,
        "mac": false,
        "quirks": false,
        "svg": true,
        "dom-contains": true,
        "css-user-select": "userSelect",
        "json-parse": true,
        "json-stringify": true,
        "event-stopimmediatepropagation": true,
        "config-useDeferredInstrumentation": false,
        "activex": false,
        "dojo-force-activex-xhr": false,
        "dojo-preload-i18n-Api": 1,
        "dojo-v1x-i18n-Api": 1,
        "esri-secure-context": false,
        "esri-wasm": true,
        "esri-workers": true,
        "esri-url-encodes-apostrophe": true,
        "esri-webgl": true,
        "esri-webgl-high-precision-fragment": true,
        "esri-webgl-vertex-shader-samplers": true,
        "esri-webgl-element-index-uint": true,
        "esri-webgl-standard-derivatives": true,
        "esri-webgl-instanced-arrays": true,
        "esri-webgl-major-performance-caveat": false,
        "esri-featurelayer-webgl": true,
        "esri-featurelayer-webgl-labeling": true,
        "esri-webgl2": true,
        "dojo-config-addOnLoad": 1,
        "dom-quirks": false,
        "events-mouseenter": true,
        "events-mousewheel": true,
        "events-keypress-typed": true,
        "config-selectorEngine": "acme",
        "dom-qsa2.1": true,
        "dom-matches-selector": true,
        "dom-qsa": true,
        "dojo-config-require": 1,
        "esri-promise-compatibility-deprecation-warnings": true,
        "dijit-legacy-requires": false,
        "dojo-bidi": false,
        "highcontrast": false,
        "esri-workers-arraybuffer-transfer": true,
        "esri-svg-classlist": true
    },
    "paths": {},
    "pathsMapProg": [],
    "packs": {
        "dojo": {
            "main": "main",
            "name": "dojo",
            "location": "./dojo",
            "lib": "."
        },
        "dijit": {
            "main": "main",
            "name": "dijit",
            "location": "./dijit",
            "lib": "."
        },
        "dojox": {
            "main": "main",
            "name": "dojox",
            "location": "./dojox",
            "lib": "."
        },
        "esri": {
            "main": "kernel",
            "name": "esri",
            "location": "./arcgis-js-api"
        },
        "@dojo": {
            "main": "main",
            "name": "@dojo",
            "location": "./@dojo",
            "lib": "."
        },
        "cldrjs": {
            "main": "dist/cldr",
            "name": "cldrjs",
            "location": "./cldrjs"
        },
        "globalize": {
            "main": "dist/globalize",
            "name": "globalize",
            "location": "./globalize"
        },
        "tslib": {
            "main": "tslib",
            "name": "tslib",
            "location": "./tslib"
        },
        "moment": {
            "main": "main",
            "name": "moment",
            "location": "./moment",
            "lib": "."
        },
        "dstore": {
            "main": "main",
            "name": "dstore",
            "location": "./dojo-dstore",
            "lib": "."
        },
        "maquette": {
            "main": "dist/maquette.umd",
            "name": "maquette",
            "location": "./maquette",
            "resourceTags": {}
        },
        "maquette-css-transitions": {
            "main": "dist/maquette-css-transitions.umd",
            "name": "maquette-css-transitions",
            "location": "./maquette-css-transitions",
            "resourceTags": {}
        },
        "maquette-jsx": {
            "main": "dist/maquette-jsx.umd",
            "name": "maquette-jsx",
            "location": "./maquette-jsx",
            "resourceTags": {}
        }
    },
    "aliases": [],
    "mapProgs": [
        ["globalize", [
            ["cldr/supplemental", "cldrjs/dist/cldr/supplemental", {}, 17],
            ["cldr/unresolved", "cldrjs/dist/cldr/unresolved", {}, 15],
            ["cldr/event", "cldrjs/dist/cldr/event", {}, 10],
            ["cldr", "cldrjs/dist/cldr", {}, 4]
        ], {}, 9]
    ],
    "modules": {},
    "cache": {},
    "locale": "en-us",
    "baseUrl": "./",
    "afterOnLoad": true
}
```

</details>


Webpack config:

<details>

```javascript
const ArcGISPlugin = require('@arcgis/webpack-plugin');
const HtmlWebPackPlugin = require('html-webpack-plugin');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const CleanWebpackPlugin = require('clean-webpack-plugin');
const path = require('path');
const MinifyPlugin = require('terser-webpack-plugin');

module.exports = {
    entry: {
        'main': './src/index.js',
    },
    output: {
        filename: '[name].[chunkhash].js',
        publicPath: '',
    },

    plugins: [
        new CleanWebpackPlugin(['dist', 'build']),
        new ArcGISPlugin({
            useDefaultAssetLoaders: false,
        }),
        new MiniCssExtractPlugin({
            // Options similar to the same options in webpackOptions.output
            // both options are optional
            filename: './[name].[hash].css',
            chunkFilename: './[id].[hash].css',
        }),
        new HtmlWebPackPlugin({
            template: './src/index.html',
            filename: './index.html',
            chunksSortMode: 'none',
        }),
        new HtmlWebPackPlugin({
            inject: false,
            template: './src/oauth-callback.html',
            filename: './oauth-callback.html',
            chunksSortMode: 'none',
        }),
    ],

    module: {
        rules: [{
            test: /\.stache$/,
            use: [{loader: 'raw-loader'}],
        }, {
            test: /\.layout$/,
            use: [{loader: 'raw-loader'}],
        }, {
            test: /\.js$/,
            exclude: /node_modules/,
            use: [{
                loader: 'babel-loader',
                options: {
                    sourceMap: true,
                },
            }],
        }, {
            test: /\.css$/,
            use: [
                MiniCssExtractPlugin.loader,
                {loader: 'css-loader', options: {sourceMap: true}},
                'postcss-loader', {
                    loader: 'resolve-url-loader',
                    options: {includeRoot: true},
                },
            ],
        }, {
            test: /\.less$/,
            use: [
                MiniCssExtractPlugin.loader,
                'css-loader',
                'postcss-loader',
                'less-loader',
            ],
        }, {
            test: /\.scss$/,
            use: [
                MiniCssExtractPlugin.loader,
                {loader: 'css-loader', options: {sourceMap: true}},
                'postcss-loader', {
                    loader: 'resolve-url-loader',
                    options: {includeRoot: true, sourceMap: true},
                }, {
                    loader: 'sass-loader',
                    options: {
                        sourceMap: true,
                    },
                },
            ],
        }, {
            test: /\.(jpe?g|png|gif|webp)$/,
            loader: 'url-loader',
            options: {
                // Inline files smaller than 10 kB (10240 bytes)
                limit: 10 * 1024,
            },
        }, {
            test: /\.(wsv|ttf|otf|eot|svg|woff(2)?)(\?[a-z0-9]+)?$/,
            use: [{
                loader: 'file-loader',
                options: {
                    name: 'build/[name].[ext]',
                },
            }],
        }],
    },


    resolve: {
        extensions: ['.wasm', '.js', '.json'],
        modules: [
            path.resolve(__dirname, '/src'),
            path.resolve(__dirname, 'node_modules/'),
        ],
        alias: {
            'esri': 'arcgis-js-api',
            'sweetalert2$': 'sweetalert2/dist/sweetalert2',
            'moment': 'moment/moment',
        },
    },


    node: {
        process: false,
        global: false,
        fs: 'empty',
    },


    externals: [
        (context, request, callback) => {
            if (/pe-wasm$/.test(request)) {
                return callback(null, 'amd ' + request);
            }
            callback();
        },
    ],

    optimization: {
        minimizer: [
            new MinifyPlugin({
                parallel: true,
            }),
        ],
    },
};

```
</details>


EsriConfig.js for workers:

<details>

```
import esriConfig from 'esri/config';

const DEFAULT_WORKER_URL = 'https://js.arcgis.com/4.9/';
const DEFAULT_LOADER_URL = `${DEFAULT_WORKER_URL}dojo/dojo-lite.js`;

esriConfig.workers.loaderUrl = DEFAULT_LOADER_URL;
esriConfig.workers.loaderConfig = {
    baseUrl: `${DEFAULT_WORKER_URL}dojo`,
    has: {
        'esri-featurelayer-webgl': 1,
    },
    packages: [
        {name: 'esri', location: `${DEFAULT_WORKER_URL}esri`},
        {name: 'dojo', location: `${DEFAULT_WORKER_URL}dojo`},
        {name: 'dojox', location: `${DEFAULT_WORKER_URL}dojox`},
        {name: 'dstore', location: `${DEFAULT_WORKER_URL}dstore`},
        {name: 'moment', location: `${DEFAULT_WORKER_URL}moment`},
        {name: '@dojo', location: `${DEFAULT_WORKER_URL}@dojo`},
        {
            name: 'cldrjs',
            location: `${DEFAULT_WORKER_URL}cldrjs`,
            main: 'dist/cldr',
        },
        {
            name: 'globalize',
            location: `${DEFAULT_WORKER_URL}globalize`,
            main: 'dist/globalize',
        },
        {
            name: 'maquette',
            location: `${DEFAULT_WORKER_URL}maquette`,
            main: 'dist/maquette.umd',
        },
        {
            name: 'maquette-css-transitions',
            location: `${DEFAULT_WORKER_URL}maquette-css-transitions`,
            main: 'dist/maquette-css-transitions.umd',
        },
        {
            name: 'maquette-jsx',
            location: `${DEFAULT_WORKER_URL}maquette-jsx`,
            main: 'dist/maquette-jsx.umd',
        },
        {name: 'tslib', location: `${DEFAULT_WORKER_URL}tslib`, main: 'tslib'},
    ],
};
```

</details>

## Related:

https://github.com/Esri/arcgis-webpack-plugin/issues/22
https://stackoverflow.com/questions/51986624/webpack-babel-loader-outputting-full-c-drive-urls-in-bundles/52007420#52007420

--- a/lib/DojoAMDModuleFactoryPlugin.js
+++ b/lib/DojoAMDModuleFactoryPlugin.js
@@ -149,7 +149,7 @@ module.exports = class DojoAMDModuleFactoryPlugin {
 				data.dependencies.some(dep => {
 					if (dep instanceof SingleEntryDependency || dep instanceof DojoAMDRequireItemDependency) {
 						if (!path.isAbsolute(data.request)) {
-							const context = dep.issuerModule && (dep.issuerModule.absMid || dep.issuerModule.request);
+							const context = dep.issuerModule && dep.issuerModule.absMid;
 							var dojoRequire = callSyncBail(this.compiler, "get dojo require");
 							const absMid = this.toAbsMid(data.request, context, dojoRequire);
 							if (absMid.charAt(0) !== '.') {

